api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceImpl.java

```diff
@@ -105,6 +106,27 @@ public void setInitialCreditsExpirationBypassed(DbUser user, boolean isBypassed)
     userInitialCreditsExpiration.setBypassed(isBypassed);
   }
 
+  @Override
+  public void extendInitialCreditsExpiration(DbUser user) {
+    DbUserInitialCreditsExpiration userInitialCreditsExpiration =
+        user.getUserInitialCreditsExpiration();
+    if (userInitialCreditsExpiration == null) {
+      throw new BadRequestException("User does not have initial credits expiration set.");
```

Comment:
does not feel like a `BadRequestException`, nothing todo with user request.  maybe an internal exception?

Reply:
That seems reasonable. I changed it to a WorkbenchExpetion which returns a 500 error.